pkg/ruler/remotequerier_test.go

```diff
@@ -785,8 +726,8 @@ func TestRemoteQuerier_StatusErrorResponses(t *testing.T) {
 				"status": "error","errorType": "execution"
 			}`),
 		}
-		error4xx = status.Error(http.StatusUnprocessableEntity, "this is a 4xx error")
-		error5xx = status.Error(http.StatusInternalServerError, "this is a 5xx error")
+		error4xx = httpgrpc.Errorf(http.StatusUnprocessableEntity, "this is a 4xx error")
+		error5xx = httpgrpc.Errorf(http.StatusInternalServerError, "this is a 5xx error")
```

Comment:
Note: these are outside the scope changes. It was just confusing, while I was reading through the code, that we use non-gRPC codes in gRPC statuses, while [their docs explicitly ask not doing that][1] (I'm now aware that `httpgrpc` has different semantics for the codes, while keeping the same implementation) ;)

[1]: https://pkg.go.dev/google.golang.org/grpc/codes